### ISSUE TITLE
List index not out of range

### ISSUE THREAD

**Issue by HolQue** [test-fullautomation/python-jsonpreprocessor#87]
I try to assign a value to a list index, that does not exist:

`${componentB_param_1}[50] : "componentB value 1.1 (new)"`

Error message:

`'Could not set variable 'componentB_param_1[50]' with value 'componentB value 1.1 (new)'!'`

The list `componentB_param_1` only contains 2 list elements. Therefore I expect an error message like:

`List index '50' out of range (maximum possible index: 1).`

**Comment by test-fullautomation** [comment]
Hi Son,

this finding is correct. Implicitely we can only create an whole array, but not a specific entry.
So it depends on the size the array had at creation time.

Thank you,
 Thomas

**Comment by namsonx** [comment]
Hello Holger, Hello Thomas,

I created the pull-request #119  for this ticket.
A reason is added in to error message in case list index out of range, an error message looks like below:
```
  File "c:\Users\mas2hc\Desktop\program\RobotFramework\python39\lib\site-packages\JsonPreprocessor\CJsonPreprocessor.py", line 505, in __jsonUpdated
    raise Exception(errorMsg)
Exception: Could not set variable 'listTest[8]' with value 'new value'! Reason: list assignment index out of range
```

Thank you,
Son

**Comment by HolQue** [comment]
Retest successful

**Comment by test-fullautomation** [comment]
solved and already merged with #147
